R/combineDatasets.R

```diff
@@ -1,93 +1,75 @@
 #' Combine sample expression profile with reference datasets
 #'
 #' Combines sample expression profile with reference datasets. Outputs a vector
-#' with first element containing the merged data and second element containing 
+#' with first element containing the merged data and second element containing
 #' merged targets info
 #'
 #' @param sample_name Sample name.
 #' @param sample_counts Sample counts.
 #' @param ref_data Reference data.
 #' @param report_dir Report directory.
-#' @param dataset Sample expression profile.
 #'
 #' @return Combined sample expression profile with reference datasets.
 #' @export
-combineDatasets <- function(sample_name, sample_counts, ref_data, report_dir, dataset) {
-
-  ##### Extract info about target file for the external reference dataset
-  target.ext <- utils::read.table(ref_data[["ext_ref"]][2], sep="\t", as.is=TRUE, header=TRUE)
-  target.ext <- cbind(target.ext, rep(ref_data[["ext_ref"]][3], nrow(target.ext)))
-  colnames(target.ext)[ncol(target.ext)] <- "Dataset"
-
-  ##### Add prexit to sample names
-  rownames(target.ext) <- paste(target.ext[,"Dataset"], target.ext[,"Sample_name"], sep = ".")
-  target.ext <- target.ext[, -1]
-
-  ##### Extract info about target file for the internal reference dataset
-  target.int <- utils::read.table(ref_data[["int_ref"]][2], sep="\t", as.is=TRUE, header=TRUE)
-  target.int <- cbind(target.int, rep(ref_data[["int_ref"]][3], nrow(target.int)))
-  colnames(target.int)[ncol(target.int)] <- "Dataset"
-
-  ##### Add prexit to sample names
-  rownames(target.int) <- paste(target.int[,"Dataset"], target.int[,"Sample_name"], sep = ".")
-  target.int <- target.int[, -1]
-
-  target.comb <- rbind(target.ext, target.int)
-
-  ##### Add sample info
-  target.sample <- data.frame(sample_name, sample_name)
-  names(target.sample) <- names(target.comb)
-  rownames(target.sample) <- sample_name
-  target.comb <- rbind( target.comb, target.sample )
-
-  ##### Make syntactically valid names
-  rownames(target.comb) <- make.names(rownames(target.comb))
+combineDatasets <- function(sample_name, sample_counts, ref_data, report_dir) {
+  assertthat::assert_that(!is.null(sample_counts), msg = "The 'sample_counts' cannot be NULL!")
+  .read_target <- function(targ, ds) {
```

Comment:
For my information, any particular reason to use `.name`?

Reply:
Not really, I guess it makes it clear this is an internal func and it also makes it hidden in the RStudio env pane when testing (which means less clutter).